### ISSUE TITLE
Bot corrups meta.yaml file of current Rust feedstock

### ISSUE THREAD

**Issue by pkgw** [regro/cf-scripts#1658]
The bot is currently corrupting the rust-feedstock meta.yaml, as seen in https://github.com/conda-forge/rust-feedstock/pull/139 .

It changes:

```
source:
  -
    url: https://static.rust-lang.org/dist/rust-{{ version }}-x86_64-unknown-linux-gnu.tar.gz  # [linux and x86_64]
    [many more ...]
```

to:

```
source:
  - url: https://static.rust-lang.org/dist/rust-{{ version }}-x86_64-unknown-linux-gnu.tar.gz  # [linux and x86_64]
    [many more ...]
```

Which means that on any platform *other* than linux-64, the list indicator is lost.

The structure of the Rust meta.yaml changed recently to add a `rust-src` source tarball, which I suspect is the culprit here.

**Comment by beckermr** [comment]
Hmmmm. I bet this is due to the selectors. I'll look into this when I get a chance. In the mean time, maybe there is a change we can make to the recipe?

**Comment by beckermr** [comment]
Yeah I bet if you make a separate list for each platform it will work. The issue is that you need the dash to have a selector too.

**Comment by pkgw** [comment]
@beckermr Yeah, I bet you're right. Is there a convenient way to test what the bot will do with a sample file?

**Comment by beckermr** [comment]
Not easily no.
